### PR TITLE
lazy init response parser

### PR DESCRIPTION
**Pull request by abhisheksarka**
### Summary
Issue: https://github.com/rails/rails/issues/34676

### CODE REVIEW

**Review by gmcgibbon — CHANGES_REQUESTED** [review]
Can you give @tbuehlmann credit since you're just applying his diff? Also can you add a test and changelog entry?

**Comment by kamipo** [comment]
Also, can you add a test case?

**Comment by abhisheksarka** [comment]
Makes sense! Will do!

File: actionpack/lib/action_dispatch/testing/test_response.rb [hunk 1]
```diff
@@ -16,7 +16,6 @@ def self.from_response(response)
 
     def initialize(*) # :nodoc:
       super
-      @response_parser = RequestEncoder.parser(content_type)
```
Comment (kaspth) [inline actionpack/lib/action_dispatch/testing/test_response.rb]: No reason to keep the initialize method after this.

**Comment by tbuehlmann** [comment]
Not sure how to feel about this. But well, go ahead.

**Comment by brandondees** [comment]
consider https://help.github.com/articles/creating-a-commit-with-multiple-authors/ ?

**Comment by kaspth** [comment]
@abhisheksarka it's great that you want to contribute, but unfortunately you've hit a somewhat unspoken truth in open source: it's not just about the code, but also how the code gets contributed.

Which brings us to how to attribute credit to a change. And sidestepping @tbuehlmann after @rafaelfranca specifically asked him to open a PR, copying his suggested patch without at least an attempt to also give credit leaves a sour taste for the PR context.

As such, I've decided to let @tbuehlmann take over in his own PR first and should that stall we can put it up for grabs. Thanks!

**Comment by abhisheksarka** [comment]
@tbuehlmann Go ahead man, my bad to begin with. @kaspth It makes perfect sense and I agree with you guys. Just started with the rails source code and thought of taking up the small stuff so came across this PR. Did not give a lot of thought to how it is being contributed. Will keep this in mind going forward

**Comment by abhisheksarka** [comment]
@tbuehlmann Go ahead man, my bad to begin with. @kaspth Just started with the rails source code and thought of taking up the small stuff so came across this PR. Did not give a lot of thought to how it is being contributed. Will keep this in mind going forward

**Pull request closed without merging**